### PR TITLE
Add parametrization with respect to hardware parameters

### PR DESCRIPTION
**Pull request by CarlottaSartore**

### CODE REVIEW

**Comment by CarlottaSartore** [comment]
This PR is still a draft since I want to implement the tests for the hardware parameters. 

@Giulero  @lrapetti  in the meanwhile if you want to start taking a look it would be great (I think we would need tons of modifications :) )

C.C. @traversaro

File: examples/iCub_example_casadi_hardware_parameters.ipynb [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+    "## Import the robot .urdf\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "urdf_path = \"/home/carlotta/iit_ws/ergocub-gazebo-simulations/models/stickBot/model.urdf\"\n",
```
Comment (CarlottaSartore) [inline examples/iCub_example_casadi_hardware_parameters.ipynb:40]: Note: this path is hardcoded I have to find a workaround on it

File: examples/iCub_example_casadi_hardware_parameters.ipynb [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from adam.casadi.computations import KinDynComputations\n",
+    "from adam.core import link_parametric \n",
+    "from adam.geometry import utils\n",
+    "import numpy as np\n",
+    "import casadi as cs\n",
+    "import gym_ignition_models"
```
Comment (traversaro) [inline examples/iCub_example_casadi_hardware_parameters.ipynb:14]: I am not sure where this is used.

Comment (CarlottaSartore) [reply examples/iCub_example_casadi_hardware_parameters.ipynb]: I have removed the import not used (which was the `gym ignition models `one), the others are needed to run the example

**Comment by CarlottaSartore** [comment]
I have added the tests for casadi implementation, @Giulero  once you have taken a look we can decide how to port these changes to the other implementations!

**Comment by CarlottaSartore** [comment]
This PR is deprecated, I have opened a new one see #31. 

Closing

**Pull request closed without merging**